### DISCUSSION TITLE
how to use auto-orient (where does it go in this command?

### DISCUSSION THREAD

**Discussion by joeldcanfield** [ImageMagick/ImageMagick#7589] [Digital Image Processing] [answered]
where would I put `-auto-orient` in this command to, well, automatically orient the output?

`magick gotg01.jpg +repage -background none -distort SRT '300,300 1.1,1.1 0' +repage zoom/gotg01-1.1.jpg`

the image is rotated 90º per the EXIF info. also open to modifying the EXIF data since this is a temporary image I'm using for a larger project. 

desired result is this image zoomed X% and oriented top up, however we get there.

**Comment by fmw42** [comment]
Right after the input

**Comment by joeldcanfield** [comment]
I only ask because I tried it in every single space in this command, including there, so something more seems to be amiss

the command

`magick gotg01.jpg -auto-orient +repage -background none -distort SRT '300,300 1.1,1.1 0' +repage zoom/gotg01-1.1.jpg`

still results in an image rotated 90º

original file attached
![gotg01](https://github.com/user-attachments/assets/4f7c31af-d615-43eb-8b80-cbf5b84b0fce)

**Reply by fmw42** [reply]
Is the image above your input or output?  I tried your command and simpler ones and it does not correct the orientation even though there is Orientation: RightTop in the meta data.  So I suspect your file is corrupt in some way or IM has a bug about correcting orientation. One of the developers or people who can look at the code might review this.

**Comment by snibgo** [comment, marked as answer]
The posted image is already correctly oriented. But the metadata says it isn't. The metadata wrongly says it needs to be rotated.

**Reply by joeldcanfield** [reply]
best way to edit the metadata? 

or better to manually rotate the image (almost certainly what happened here) so the visible orientation matches the exif Orientation value?

**Reply by fmw42** [reply]
Exiftool.  https://exiftool.org

**Comment by joeldcanfield** [comment]
indeed; thought of that five minutes ago and had to reinstall. 

thanks again.

**Comment by snibgo** [comment]
> best way to edit the metadata?

The IM operation "`-orient undefined`" will remove the bad metadata.

**Comment by joeldcanfield** [comment]
I'm guessing from the wording that nukes it rather than correcting it, yes? 

I used exiftool to change the Orientation on the half dozen images giving me trouble and all seems well.
